tests/testutils/addon_helpers.py

```diff
@@ -69,10 +69,9 @@ def addon_with_indeximage():
 @pytest.fixture
 def addons_managed_by_addon_cr():
     def create_addon_managed_by_addon_cr(path):
-        addon = Addon(path, "stage")
-        addon.manager = AddonManager.ADDON_OPERATOR
-        addon.sss = Sss(addon=addon)
-        return addon
+        return Addon(
+            path, "stage", override_manager=AddonManager.ADDON_OPERATOR
```

Comment:
I am not a big fan of adding an function param simply to accomodate testing. Would suggest doing:

```python
addon = Addon(path, "stage")
addon.manager = AddonManager.ADDON_OPERATOR
return addon
```
```

Reply:
That would actually lead to unexpected results because
when the first line would execute, the `Addon(...)` constructor would initialise `addon.sss` assuming the manager to be `UNKNOWN` ([Ref](https://github.com/mt-sre/managed-tenants-cli/blob/main/managedtenants/core/addons_loader/addon.py#L56-L61
)) hence, the rendering the [wrong section](https://github.com/mt-sre/managed-tenants-cli/blob/main/managedtenants/data/selectorsyncset.yaml.j2#L9-L19) of the SSS Jinja template.

In fact, to counter this issue, till now (before this PR), the code was explicitly re-setting the `addon.sss` by doing the following 
```python
        addon.manager = AddonManager.ADDON_OPERATOR
        addon.sss = Sss(addon=addon)
```

Ref: https://github.com/mt-sre/managed-tenants-cli/blob/main/tests/testutils/addon_helpers.py#L73-L74

This was functionally correct but this was unnecessarily executing the SSS Walk execution twice (which happens everytime the `Sss()` constructor is called)

> Just to re-iterate: it will be called at `addon = Addon(...)` and `addon.sss = Sss(addon=addon)`

Hence, I felt the `Addon()` constructor should allow to provide/override the addon manager. 
And the `override_manager` argument has a default value (None) as well set in the [constructor's signature](https://github.com/mt-sre/managed-tenants-cli/blob/c2ce4d90421ddf88d0cf7ebb96fd02ac41fdc8f1/managedtenants/core/addons_loader/addon.py#L27) hence, doesn't require any change or even awareness about this change across the codebase.

Reply:
You are right, we've been rendering the Sss twice and my snippet was missing this info.